boilerplate/app/components/Toggle.tsx

```diff
@@ -0,0 +1,493 @@
+    { tintColor: colors.palette.neutral100 },
+    disabled && { tintColor: colors.palette.neutral600 },
+    $checkboxIconStyleOverride,
+  ]
+
+  return (
+    <Animated.View style={$checkboxContainerStyle}>
+      <Image source={iconRegistry.check} style={$checkboxIconStyle} />
+    </Animated.View>
+  )
+}
+
+function Radio(props: ToggleInputProps) {
+  const {
+    on,
+    status,
+    disabled,
+    innerStyle: $radioContainerStyleOverride,
+    detailStyle: $radioDotStyleOverride,
+  } = props
+
+  const $animatedRadioContainer = useAnimatedStyle(
+    () => ({ opacity: withTiming(on ? 1 : 0) }),
+    [on],
+  )
+
+  const $radioContainerStyle = [
+    $radioContainer,
+    { backgroundColor: colors.palette.neutral100 },
+    disabled && { backgroundColor: colors.transparent },
+    $animatedRadioContainer,
+    $radioContainerStyleOverride,
+  ]
+  const $radioDotStyle = [
+    $radioDot,
+    { backgroundColor: colors.palette.secondary500 },
+    status === "error" && { backgroundColor: colors.error },
+    disabled && { backgroundColor: colors.palette.neutral600 },
+    $radioDotStyleOverride,
+  ]
+
+  return (
+    <Animated.View style={$radioContainerStyle}>
+      <View style={$radioDotStyle} />
+    </Animated.View>
+  )
+}
+
+function Switch(props: ToggleInputProps) {
+  const {
+    on,
+    status,
+    disabled,
+    innerStyle: $switchContainerStyleOverride,
+    detailStyle: $switchKnobStyleOverride,
+  } = props
+  const [renderedKnobWidth, setRenderedKnobWidth] = useState(0)
+
+  const $animatedSwitchContainer = useAnimatedStyle(() => {
+    const onBackgroundColor = (function () {
+      if ($switchContainerStyleOverride?.backgroundColor) {
+        return $switchContainerStyleOverride.backgroundColor
+      }
+      if (disabled) return colors.transparent
+      if (status === "error") return colors.error
+      return colors.palette.secondary500
+    })()
+
+    const backgroundColor = withTiming(on ? onBackgroundColor : colors.transparent)
+
+    return { backgroundColor }
+  }, [status, disabled, on])
+
+  const $animatedSwitchKnob = useAnimatedStyle(() => {
+    const offBackgroundColor = (function () {
+      if ($switchContainerStyleOverride?.backgroundColor) {
+        return $switchContainerStyleOverride.backgroundColor
+      }
+      if (disabled) return colors.palette.neutral600
+      if (status === "error") return colors.error
+      return colors.palette.secondary500
+    })()
+
+    const onBackgroundColor = (function () {
+      if ($switchKnobStyleOverride?.backgroundColor) return $switchKnobStyleOverride.backgroundColor
+      if (disabled) return colors.palette.neutral600
+      return colors.palette.neutral100
+    })()
+
+    const backgroundColor = withTiming(on ? onBackgroundColor : offBackgroundColor)
+    const start = withTiming(on ? "100%" : "0%")
+    const marginStart = withTiming(on ? -renderedKnobWidth : 0)
+
+    return { backgroundColor, start, marginStart }
+  }, [status, disabled, on])
```

Comment:
Animation for the knob. The position (`start`) could have been done simpler. But since we support customization (including sizes), we can't animate between static ranges (e.g. `0, 25`). Instead, we animate from 0 to 100 and offset the knob by it's rendered-size using `marginStart`. Theoretically, we can pull it's size from the defined styles, but since a user can customize with a string value, this becomes a bit more complex. Instead we just measure it and use its point value here.